server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -581,13 +560,14 @@ public void run() {
     }
   }
 
-  private boolean shouldCleanupMigration(TabletMetadata tabletMetadata) {
+  boolean shouldCleanupMigration(TabletMetadata tabletMetadata) {
     var tableState = getContext().getTableManager().getTableState(tabletMetadata.getTableId());
     var migration = tabletMetadata.getMigration();
     Preconditions.checkState(migration != null,
         "This method should only be called if there is a migration");
     return tableState == TableState.OFFLINE || !onlineTabletServers().contains(migration)
-        || tabletMetadata.getLocation().getServerInstance().equals(migration);
+        || (tabletMetadata.getLocation() != null
```

Comment:
This is the 2nd existing bug that was fixed in this change (the first was fetching the location column when calling this).